### PR TITLE
Add Theta blockchain support

### PR DESCRIPTION
**Pull request by growlot**
## Description

- Add send transaction
- Add signer
- Add unit test for Transaction
- Add unit test for Signer
- Add TWCoinType
- Add curve and purpose
- Add address validation and derivation
- Add coin configuration
- Add signing interface
- Add unit test for swift project
- Add coin icon into README.md

## Testing instructions
Run the unit test in tests/Theta

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist
- [x] Implement functionality in C++. Put it in a subfolder of `src/`.
    - [ ] Address (Use Ethereum Address instead)
    - [x] Transaction
    - [x] Signer
- [x] Write unit tests. Put them in a subfolder of `tests/`.
- [x] Add relevant constants in `TWCoinType`, `TWP2SHPrefix`, `TWEthereymChainID`, `TWHRP`, etc., as necessary.
- [x] Return correct curve and purpose in `src/Coin.cpp`.
- [x] Implement address validation and derivation in `src/Coin.cpp`.
- [x] Implement coin configuration `src/include/TWCoinTypeConfiguration.cpp`.
- [x] Write interface header in `include/TrustWalletCore` and implement the interface in `src/interface`.
    - [ ] Address interface (Use Ethereum Address interface instead).
    - [x] Signing interface.
- [x] Validate generated code in Android an iOS projects. Write integration tests for each.
- [x] Add a coin icon to "Supported Blockchains" section in `README.md`
    - [x] Upload coin icon to [TrustWallet/tokens](https://github.com/TrustWallet/tokens) if necessary

### CODE REVIEW

**Comment by hewigovens** [comment]
could you please rebase master?

File: src/Theta/Signer.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    auto payload = Data();
+    append(payload, RLP::encode(chainID));
+    append(payload, transaction.encode());
+    append(encoded, RLP::encode(payload));
+    return RLP::encodeList(encoded);
+}
+
+Data Signer::sign(const PrivateKey& privateKey, const Transaction& transaction) noexcept {
+    auto encoded = dataForSign(transaction);
+    std::cerr << hex(encoded) << std::endl;
```
Comment (alejandro-isaza) [inline src/Theta/Signer.cpp]: Please remove.

File: README.md [hunk 2]
```diff
@@ -35,6 +35,7 @@ Trust Wallet Core is a cross-platform library that implements low-level cryptogr
 <a href="https://thundercore.com" target="_blank"><img src="https://raw.githubusercontent.com/TrustWallet/tokens/master/coins/1001.png" width="32" /></a>
 <a href="https://kinecosystem.org/" target="_blank"><img src="https://raw.githubusercontent.com/TrustWallet/tokens/master/coins/2017.png" width="32" /></a>
 <a href="https://decred.org/" target="_blank"><img src="https://raw.githubusercontent.com/TrustWallet/tokens/master/coins/42.png" width="32" /></a>
+<a href="https://www.thetatoken.org/" target="_blank"><img src="https://raw.githubusercontent.com/TrustWallet/tokens/master/coins/theta.png" width="32" /></a>
```
Comment (hewigovens) [inline README.md]:
```suggestion
<a href="https://www.thetatoken.org/" target="_blank"><img src="https://raw.githubusercontent.com/TrustWallet/tokens/master/coins/500.png" width="32" /></a>
```

File: src/Theta/Signer.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+#include "../HexCoding.h"
+#include "../Ethereum/RLP.h"
+#include "../Hash.h"
+
+using namespace TW;
+using namespace TW::Theta;
+using RLP = Ethereum::RLP;
+
+Data Signer::dataForSign(const Transaction& transaction) noexcept {
```
Comment (hewigovens) [inline src/Theta/Signer.cpp]:
```suggestion
Data Signer::encode(const Transaction& transaction) noexcept {
```

File: src/Theta/Transaction.cpp [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    append(encoded, encode(input.coins));
+    append(encoded, RLP::encode(input.sequence));
+    append(encoded, RLP::encode(input.signature));
+    return RLP::encodeList(encoded);
+}
+
+Data encode(const std::vector<TxInput>& inputs) noexcept {
+    auto encoded = Data();
+    for (const auto& input : inputs)
+    {
```
Comment (hewigovens) [inline src/Theta/Transaction.cpp]: could you please run `clang-format` and follow existing code style?

File: src/Theta/Signer.cpp (hunk 1 above)
Comment (growlot) [reply src/Theta/Signer.cpp]: I missed to remove it. Let me do. Thank you.

File: README.md (hunk 2 above)
Comment (growlot) [reply README.md]: Will do, thanks.

File: src/Theta/Signer.cpp (hunk 3 above)
Comment (growlot) [reply src/Theta/Signer.cpp]: Good suggestion, thank you.

File: src/Theta/Transaction.cpp (hunk 4 above)
Comment (growlot) [reply src/Theta/Transaction.cpp]: Yep let me do. Thank you.

**Pull request merged**